src/components/Heading/Heading.tsx

```diff
@@ -172,20 +181,60 @@ export const Heading = chakra(
             }
           : { fontSize: `mobile.heading.${finalRoot}${sizeIndex}` }
         : undefined;
-      return (
-        <ChakraHeading
-          as={asHeading}
-          className={className}
-          id={id}
-          ref={ref}
-          sx={{
-            ...styles,
-            ...responsiveStyles,
-          }}
-          {...rest}
-        >
-          {content}
-        </ChakraHeading>
+      const overlineSize = !isNaN(sizeIndex)
```

Comment:
I feel that the coding from here thru line 231 could be more concise. What is here now does work, but it seems long-winded.

Reply:
Per @EdwinGuzman:
It does seem long-winded but I don't see an easy way to make it more concise. It's not apparent but I can take a closer look soon.